### ISSUE TITLE
64 bit?

### ISSUE THREAD

**Issue by jonnyijapan** [gibtang/CCNSCoding#1]
The bundled version of Cocos2d-x does not compile/run on 64 bit devices. I managed to get it all running on an iPhone 5.

So maybe you want to update this to run on 64-bit devices out of the box...

I guess the CCNSCoding related classes should run in any setup though, I will give that a spin from now. Thanks.

**Comment by IgorVovk** [comment]
Yes, this example based on outdated version of cocos2dx, actually you don't need it, you can setup project using any version of cocos2dx and the add to Classes 
libs/rapidjson/\* 
utilities/CCNSCoder.cpp  
utilities/CCNSCoder.h

Since at least somebody started using it I have to upload more advanced version with tests and using examples. Will try to do it over the weekend.

Thank you.

**Comment by jonnyijapan** [comment]
Ok, I will get back to this issue from now. I think recent cocos2d-x already uses rapidjson - not sure if there will be any version mismatch here but I'll start out only adding these files, see what happens:

CCNSCoder.cpp
CCNSCoder.h
FileReader.cpp
FileReader.h
StringUtilities.cpp
StringUtilities.h

**Comment by IgorVovk** [comment]
Ok, you can. I just checked cocos2dx v 3.3 contains rapidjson of version "Version 0.11" which is the same as mine so there should not be any issues.

Unfortunately last weekend I got no time to prepare and upload latest version so I will try to do it asap.
Thanks.

**Comment by jonnyijapan** [comment]
Somehow leaving out your rapidjson didn't work. I just added it to the project, and I also needed to edit the reading part of FileReader beacuse it wouldn't compile, to:

```
// unsigned long file_size = 0;

ssize_t file_size;
fileContents = FileUtils::getInstance()->getFileData(fullPath, "r", &file_size);
```

And then it seems to work after one or two tests. Will see how things work when I start using this for real.
I never used NSCoding on the Cocos2D scenes, just the graphic-less background model classes. I guess that won't be a problem.

**Comment by IgorVovk** [comment]
Sending you updated version of FileReader I used with cocos2dx 3.3
https://drive.google.com/file/d/0B9QkIenAZZdYamY3bEhqeUltTnc/view?usp=sharing

I also never tried to use NSCoder for class derived from cocos2dx classes, I suppose there will be some issues as CCNSCoder is a bit tricky in creation.
